flaml/model.py

```diff
@@ -736,28 +734,18 @@ def _select_checkpoint(self, trainer):
     def _compute_metrics_by_dataset_name(self, eval_pred):
         if isinstance(self._metric, str):
             from .ml import metric_loss_score
-            from .nlp.utils import postprocess_text
+            from .nlp.utils import postprocess_prediction, postprocess_text
 
             predictions, labels = eval_pred
+            predictions = postprocess_prediction(
+                self._task, predictions, self.tokenizer
+            )
             if self._task in NLG_TASKS:
-                if isinstance(predictions, tuple):
-                    predictions = np.argmax(predictions[0], axis=2)
-                decoded_preds = self.tokenizer.batch_decode(
-                    predictions, skip_special_tokens=True
-                )
                 labels = np.where(labels != -100, labels, self.tokenizer.pad_token_id)
                 decoded_labels = self.tokenizer.batch_decode(
                     labels, skip_special_tokens=True
                 )
-                predictions, labels = postprocess_text(decoded_preds, decoded_labels)
-            else:
-                predictions = (
-                    np.squeeze(predictions)
-                    if self._task == SEQREGRESSION
-                    else np.argmax(predictions, axis=2)
-                    if self._task == TOKENCLASSIFICATION
-                    else np.argmax(predictions, axis=1)
-                )
+                predictions, labels = postprocess_text(predictions, decoded_labels)

```

Comment:
What's the difference between postprocess_text and postprocess_prediction? Also, why is line 743-748 not part of  postprocess_prediction?

Reply:
renamed.

Also, why is line 743-748 not part of postprocess_prediction?
>> because postprocess_prediction is for postprocessing the prediction, that's the code sharing part of .predict and ._compute_metrics_by_dataset_name.